flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -933,7 +940,11 @@ class ExprLowering {
           addr = builder.createTemporary(getLoc(), val.getType());
           builder.create<fir::StoreOp>(getLoc(), val, addr);
         }
-        argTypes.push_back(addr.getType());
+        if (isCharacterType(*arg))
+          argTypes.push_back(
+              fir::BoxCharType::get(builder.getContext(), /*FIXME*/ 1));
+        else
+          argTypes.push_back(addr.getType());
         operands.push_back(addr);
```

Comment:
Shouldn't addr be emboxed right also ?
I think #37 could be reverted if the only reason it was done was to get boxchar in cases like here.
Instead, one could use here:
```
auto char = builder.materializeCharacter(addr);
auto box = builder.createEmboxChar(char.first, char.second);
```
Maybe this pattern should be its own helper function that does nothing if `addr` is already a fir.boxchar.

Reply:
If I am reading your question right, then a general argument in F77 is not boxed, no. The only descriptor in play is for CHARACTER type arguments. The calling side provides no other information.

Reply:
FWIW, #37 was already reverted.